### PR TITLE
Update some dependencies and move to node20

### PR DESCRIPTION
**Pull request by DriesSchaumont**
Updated:

- [x] actions/checkout@v3 to actions/checkout@v4
- [x] actions/cache@v3 to actions/cache@v4
- [x] Updated node version to `node20`
- [x] Ran `npm update` on ns-build
- [x] Ran `npm update` on ns-list

### CODE REVIEW

**Comment by DriesSchaumont** [comment]
❗  After merging this, we need to change `viash-io/viash-actions/ns-build@update_dependencies` to `viash-io/viash-actions/ns-build@v5`.

**Comment by DriesSchaumont** [comment]
Example run here: https://github.com/openpipelines-bio/openpipeline/actions/runs/7656606163

**Review by rcannood — APPROVED** [review]
LGTM!

File: pro/build-nextflow-params/action.yml [hunk 1]
```diff
@@ -34,22 +34,22 @@ runs:
...
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
       with: 
         repository: viash-io/viash_tools
         token: ${{ inputs.viash_pro_token }}
         path: viash_tools
         ref: ${{ steps.latestrelease.outputs.releasetag }}
 
-    - uses: viash-io/viash-actions/ns-list@v3
+    - uses: viash-io/viash-actions/ns-list@update_dependencies
```
Comment (rcannood) [inline pro/build-nextflow-params/action.yml:44]:
```suggestion
    - uses: viash-io/viash-actions/ns-list@v5
```

File: pro/build-nextflow-params/action.yml [hunk 2]
```diff
@@ -34,22 +34,22 @@ runs:
...
       id: nslistcomponents
       if: inputs.components
       with:
         src: ${{ inputs.components }}
         format: "json"
         platform: "nextflow"
 
-    - uses: viash-io/viash-actions/ns-list@v3
+    - uses: viash-io/viash-actions/ns-list@update_dependencies
```
Comment (rcannood) [inline pro/build-nextflow-params/action.yml:52]:
```suggestion
    - uses: viash-io/viash-actions/ns-list@v5
```

File: pro/build-nextflow-schemas/action.yml [hunk 3]
```diff
@@ -34,22 +34,22 @@ runs:
...
-    - uses: actions/checkout@v3
+    - uses: actions/checkout@v4
       with: 
         repository: viash-io/viash_tools
         token: ${{ inputs.viash_pro_token }}
         path: viash_tools
         ref: ${{ steps.latestrelease.outputs.releasetag }}
 
-    - uses: viash-io/viash-actions/ns-list@v3
+    - uses: viash-io/viash-actions/ns-list@update_dependencies
```
Comment (rcannood) [inline pro/build-nextflow-schemas/action.yml:44]:
```suggestion
    - uses: viash-io/viash-actions/ns-list@v5

**Review by rcannood — APPROVED** [review]

**Pull request merged**